source/azure_iot.c

```diff
@@ -0,0 +1,438 @@
+    az_span xValueSpan = az_span_create( ( uint8_t * ) pucValue, ( int32_t ) ulValueLength );
+    az_result xResult;
+
+    if( pxMessageProperties == NULL )
+    {
+        AZLogError( ( "AzureIoTMessagePropertiesAppend failed: Invalid argument" ) );
+        return eAzureIoTInvalidArgument;
+    }
+
+    xResult = az_iot_message_properties_append( &pxMessageProperties->_internal.xProperties,
+                                                xNameSpan, xValueSpan );
+
+    if( az_result_failed( xResult ) )
+    {
+        return eAzureIoTFailed;
+    }
+
+    return eAzureIoTSuccess;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTResult_t AzureIoT_MessagePropertiesFind( AzureIoTMessageProperties_t * pxMessageProperties,
+                                                 const uint8_t * pucName,
+                                                 uint32_t ulNameLength,
+                                                 const uint8_t ** ppucOutValue,
+                                                 uint32_t * pulOutValueLength )
+{
+    az_span xNameSpan = az_span_create( ( uint8_t * ) pucName, ( int32_t ) ulNameLength );
+    az_span xOutValueSpan;
+    az_result xResult;
+
+    if( ( pxMessageProperties == NULL ) ||
+        ( ppucOutValue == NULL ) || ( pulOutValueLength == NULL ) )
+    {
+        AZLogError( ( "AzureIoTMessagePropertiesFind failed: Invalid argument" ) );
+        return eAzureIoTInvalidArgument;
+    }
+
+    xResult = az_iot_message_properties_find( &pxMessageProperties->_internal.xProperties,
+                                              xNameSpan, &xOutValueSpan );
+
+    if( az_result_failed( xResult ) )
+    {
+        return eAzureIoTItemNotFound;
+    }
+
+    *ppucOutValue = az_span_ptr( xOutValueSpan );
+    *pulOutValueLength = ( uint32_t ) az_span_size( xOutValueSpan );
+
+    return eAzureIoTSuccess;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTResult_t AzureIoT_Base64HMACCalculate( AzureIoTGetHMACFunc_t xAzureIoTHMACFunction,
+                                               const uint8_t * pucKey,
+                                               uint32_t ulKeySize,
+                                               const uint8_t * pucMessage,
+                                               uint32_t ulMessageSize,
+                                               uint8_t * pucBuffer,
+                                               uint32_t ulBufferLength,
+                                               uint8_t * pucOutput,
+                                               uint32_t ulOutputSize,
+                                               uint32_t * pulOutputLength )
+{
+    AzureIoTResult_t xStatus;
+    uint8_t * pucHashBuf;
+    uint32_t ulHashBufSize = AZURE_BASE64_HASH_BUFFER_SIZE;
+    uint32_t ulBinaryKeyBufSize;
+
+    if( ( xAzureIoTHMACFunction == NULL ) ||
```

Comment:
yea for now since this is public we check. the others are static so no go on those. These will be properly updated if we change the visibility of the function